### ISSUE TITLE
Authentication using an API Key

### ISSUE THREAD

**Issue by ryan-pixel** [burnash/gspread#753]
I was wondering if it was possible to authenticate with gspread using a [API key](https://cloud.google.com/docs/authentication/api-keys#creating_an_api_key). I'm trying to use AWS Lambda and I'm being prompted to open a browser page to authenticate.

**Comment by burnash** [comment]
It's funny that I'm actually looking at it this very moment. Looks like it's a recent addition and could be the simplest way to authenticate if I'm not mistaken. 

What are you currently using to auth?

**Comment by burnash** [comment]
A deeper look revealed that the keys are limited to public data:

From Sheets API Guides > [Acquiring and using an API key](https://developers.google.com/sheets/api/guides/authorizing#APIKey):

> Every request your application sends to the Google Sheets API needs to identify your application to Google. There are two ways to identify your application: using an OAuth 2.0 token (which also authorizes the request) and/or using the application's API key. Here's how to determine which of those options to use:
> * If the request requires authorization (such as a request for an individual's private data), then the application must provide an OAuth 2.0 token with the request. The application may also provide the API key, but it doesn't have to.
> * If the request doesn't require authorization (such as a request for public data), then the application must provide either the API key or an OAuth 2.0 token, or both—whatever option is most convenient for you.

Is it your case?

**Comment by ryan-pixel** [comment]
Not exactly sure. I'm just having my code access a Spreadsheet that the service account is shared with. It'll add/delete individual sheets and populate the cells with the data I need. I was trying to gspread in AWS Lambda and was running into issues with an OAuth page popping up, but I've since found a fix for it.

`
import gspread
from google.oauth2 import service_account
from google.auth.transport.requests import AuthorizedSession

credentials = service_account.Credentials.from_service_account_file('creds.json')
scoped_credentials = credentials.with_scopes(
['https://.....])

gc = gspread.Client(auth=scoped_credentials)
gc.session = AuthorizedSession(scoped_credentials)
`

The gspread part of my code seems to work, but I've run into an issue using Google Scripts API with an OAuth screen popping up, but that's a separate issue from gspread. I'm trying to use that API key I linked above with it, but am having issues. If I find a solution and you think it'll help with easy authentication with gspread, I'll gladly tell you what I did.
